sequencing/ATACSeq/jobSubmission/2_batchCalcQCMetrics.sh

```diff
@@ -6,46 +6,68 @@
+## EXECUTION: sbatch --array= ./sequencing/ATACSeq/jobSubmission/2_batchCalcQCMetrics.sh <config file>                ||
+## - execute from scripts directory                                                                                   ||
+##                                                                                                                    ||
+## INPUTS:                                                                                                            || 
+## --array -> It splits the samples into groups of 10 to run in parallel, so batch number should be the number of     ||
+##  samples divided by 10. e.g. If there are 50 samples, batch number should be 0-5, producing 5 batches of 10 samples|| 
+##  each. If the number of samples is less than 10, set batch number to 0                                             ||
+## - $1 -> <config file> : directory to config file for project                                                       ||
+##                                                                                                                    ||
+## DESCRIPTION: This script uses the ATACseqQC R package to generate the fragment distribution and calculate some     ||
+## summary statistics.                                                                                                ||
+##                                                                                                                    ||
+## ===================================================================================================================##
+
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
 
 ## print start date and time
 echo Job started on:
 date -u
-	
-## needs to be executed from the scripts folder
-echo "Changing Folder to: "
-echo $SLURM_SUBMIT_DIR
-
 
+## load config file provided on command line related to the specified project
+source $1
 echo "Loading config file for project: " $1
-export PROJECT=$1
-source ./ATACSeq/config/config.txt 
-
 echo "Project directory is: " $DATADIR
 
+## Log files directory
+LOG_DIR=ATACSeq/logFiles/${USER}/${SLURM_ARRAY_JOB_ID}
+echo "Log files will be moved to dir: " $LOG_DIR
+mkdir -p $LOG_DIR
+mv "./ATACSeq/logFiles/${USER}/ATACcalcQCS2-${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}*" $LOG_DIR
 
 ##check array specified and exit if not
 if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
 then 
     { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
 fi
 
-#-----------------------------------------------------------------------#
+##  ==================  ##
+##     CALCQCMetrics    ##
+##  ==================  ##
 
 mkdir -p ${ALIGNEDDIR}/QCOutput
 
-module load R/3.6.3-foss-2020a
-
-Rscript ${SCRIPTDIR}/ATACSeq/preprocessing/fragmentDistribution.r ${ALIGNEDDIR} ${SLURM_ARRAY_TASK_ID} 
-
-echo 'EXITCODE: ' $?
+module load R/4.2.1-foss-2022a
```

Comment:
I've had a look at the associated R script. It should still work for R version 4.2.1. Cool

Reply:
I also need to update the associated scripts